include/seqan3/alignment/cigar_conversion/cigar_from_alignment.hpp

```diff
@@ -0,0 +1,236 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2022, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2022, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the function seqan3::cigar_from_alignment and a helper struct seqan3::cigar_clipped_bases.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <vector>
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+#include <seqan3/alphabet/gap/gap.hpp>
+#include <seqan3/utility/views/zip.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Helper struct to specialise soft and hard clipping when using seqan3::cigar_from_alignment.
+ * \ingroup cigar_conversion
+ *
+ * A CIGAR string might have hard or soft clipping at the front or back of a it.
+ * E.g. `2H3S100M3S2H`.
+ */
+struct cigar_clipped_bases
+{
+    uint32_t hard_front{}; //!< The number of hard clipped bases at the front of the CIGAR string.
+    uint32_t hard_back{};  //!< The number of hard clipped bases at the back of the CIGAR string.
+    uint32_t soft_front{}; //!< The number of soft clipped bases at the front of the CIGAR string.
+    uint32_t soft_back{};  //!< The number of soft clipped bases at the back of the CIGAR string.
+};
+
+/*!\brief Creates a cigar string (SAM format) given a seqan3::detail::pairwise_alignment represented by two
+ *        seqan3::aligned_sequence's.
```

Comment:
Before you wrote:
```suggestion
 *        `seqan3::aligned_sequence`s.
```
I don't know what is better, but if you want to keep this, than you should change above.

Reply:
Here it looks like this:
![Bildschirmfoto 2022-10-18 um 09 53 31](https://user-images.githubusercontent.com/5746285/196370443-bdbeea3a-2196-442b-9941-3f550dfa7c55.png)
normally it looks like
![Bildschirmfoto 2022-10-18 um 09 53 55](https://user-images.githubusercontent.com/5746285/196370505-51eee430-6eae-486a-910e-8638bc3c817c.png)
But we probably have to accept this.